pkg/gui/context/menu_context.go

```diff
@@ -111,6 +113,30 @@ func (self *MenuViewModel) GetDisplayStrings(_startIdx int, _length int) [][]str
 	})
 }
 
+func (self *MenuViewModel) GetNonModelItems() []NonModelItem {
+	// Don't display section headers when we are filtering. The reason is that
+	// filtering changes the order of the items (they are sorted by best match),
+	// so all the sections would be messed up.
+	if self.FilteredListViewModel.IsFiltering() {
+		return []NonModelItem{}
+	}
+
+	result := []NonModelItem{}
+	menuItems := self.FilteredListViewModel.GetItems()
+	prevSection := ""
+	for i, menuItem := range menuItems {
+		if menuItem.Section != "" && menuItem.Section != prevSection {
+			result = append(result, NonModelItem{
+				Index: i,
+				Text:  theme.MenuSectionHeaderTextStyle.Sprintf("--- %s ---", menuItem.Section),
```

Comment:
This is very ad hoc for now. I'm not a good UI designer, someone please tell me what this should look like. 😄 

I could imagine that we want to align the section header with the second column of the menu items (the label); or maybe we want to center the section headers horizontally. Either way, if we are drawing with a background color, we probably want the string to be padded to the viewport width. We could pass this easily into the `getNonModelItems` function.

Reply:
Aligning with the second column might look best, but it's tricky because this is generic menu code, so we'd somehow have to pass the column to align to, maybe by making the MenuItem's Section field a struct that has a text and a column index to align to, or something like that. I'm not sure it's worth it.

For now I pushed a commit that centers the section headers horizontally.

Reply:
Here's the (hacky) patch I'm using for that:
```diff
diff --git a/pkg/gui/context/menu_context.go b/pkg/gui/context/menu_context.go
index 75f692922..cfbae5df6 100644
--- a/pkg/gui/context/menu_context.go
+++ b/pkg/gui/context/menu_context.go
@@ -5,7 +5,6 @@ import (
 	"github.com/jesseduffield/lazygit/pkg/gui/style"
 	"github.com/jesseduffield/lazygit/pkg/gui/types"
 	"github.com/jesseduffield/lazygit/pkg/utils"
-	"github.com/mgutz/str"
 	"github.com/samber/lo"
 )
 
@@ -126,9 +125,16 @@ func (self *MenuViewModel) GetNonModelItems(viewPortWidth int) []NonModelItem {
 	prevSection := ""
 	for i, menuItem := range menuItems {
 		if menuItem.Section != "" && menuItem.Section != prevSection {
+			if prevSection != "" {
+				result = append(result, NonModelItem{
+					Index: i,
+					Text:  "",
+				})
+			}
+
 			result = append(result, NonModelItem{
 				Index: i,
-				Text:  str.Pad(" "+menuItem.Section+" ", "-", viewPortWidth),
+				Text:  "         " + style.FgGreen.SetBold().Sprint("--- "+menuItem.Section+" ---"),
 			})
 			prevSection = menuItem.Section
 		}
diff --git a/pkg/gui/menu_panel.go b/pkg/gui/menu_panel.go
index 9f3b3a55d..88095584d 100644
--- a/pkg/gui/menu_panel.go
+++ b/pkg/gui/menu_panel.go
@@ -1,7 +1,8 @@
 package gui
 
 import (
-	"github.com/jesseduffield/lazygit/pkg/gui/presentation"
+	"fmt"
+
 	"github.com/jesseduffield/lazygit/pkg/gui/types"
 	"github.com/jesseduffield/lazygit/pkg/theme"
 	"github.com/jesseduffield/lazygit/pkg/utils"
@@ -27,7 +28,7 @@ func (gui *Gui) createMenu(opts types.CreateMenuOptions) error {
 		}
 
 		if item.OpensMenu {
-			item.LabelColumns[0] = presentation.OpensMenuStyle(item.LabelColumns[0])
+			item.LabelColumns[0] = fmt.Sprintf("%s...", item.LabelColumns[0])
 		}
 
 		maxColumnSize = utils.Max(maxColumnSize, len(item.LabelColumns))
diff --git a/pkg/gui/presentation/menu.go b/pkg/gui/presentation/menu.go
deleted file mode 100644
index c43896c22..000000000
--- a/pkg/gui/presentation/menu.go
+++ /dev/null
@@ -1,7 +0,0 @@
-package presentation
-
-import "github.com/jesseduffield/lazygit/pkg/gui/style"
-
-func OpensMenuStyle(str string) string {
-	return style.FgMagenta.Sprintf("%s...", str)
-}

```

I've removed the magenta colour on menu items that open a menu because I find it's distracting (but I've kept the ellipsis)

Reply:
Thanks for the design. I pushed a few fixups to implement it in a less hacky way. (Passing the column positions to `getNonModelItems` to make the alignment possible.)